common/rakelib/utils.rake

```diff
@@ -121,19 +122,26 @@ class Utils
   private_class_method :plain
 
   # select the xcode version we want/support
-  def self.version_select
-    version = '8.*'
-    xcodes = `mdfind "kMDItemCFBundleIdentifier = 'com.apple.dt.Xcode' && kMDItemVersion = '#{version}'"`.chomp.split("\n")
-    if xcodes.empty?
-      raise "\n[!!!] SwiftGen requires Xcode #{version}, but we were not able to find it. If it's already installed update your Spotlight index with 'mdimport /Applications/Xcode*'\n\n"
+  @@version_select ||= begin
+    # if current Xcode already fulfills min version don't force DEVELOPER_DIR=…
+    current_xcode = Pathname.new(`xcode-select -p`).parent.parent
+    current_xcode_version = `mdls -name kMDItemVersion -raw #{current_xcode}`.chomp
+    if current_xcode_version.to_f >= MIN_XCODE_VERSION
+      ""
+    else
+      # Get all available Xcodes, order by version, get the latest one
+      xcodes = `mdfind "kMDItemCFBundleIdentifier = 'com.apple.dt.Xcode'"`.chomp.split("\n")
+      versions = xcodes.map { |path| { :vers => `mdls -name kMDItemVersion -raw "#{path}"`, :path => path } }
+      latest_xcode = versions.sort { |p1, p2| p1[:vers] <=> p2[:vers] }.last
+
+      # Check if it's at least the right version
+      unless latest_xcode[:vers].to_f >= MIN_XCODE_VERSION
+        raise "\n[!!!] SwiftGen requires Xcode #{MIN_XCODE_VERSION}, but we were not able to find it. If it's already installed update your Spotlight index with 'mdimport /Applications/Xcode*'\n\n"
+      end
+
+      %Q(DEVELOPER_DIR="#{latest_xcode[:path]}/Contents/Developer")
     end
-    
-    # Order by version and get the latest one
-    vers = lambda { |path| `mdls -name kMDItemVersion -raw "#{path}"` }
-    latest_xcode_version = xcodes.sort { |p1, p2| vers.call(p1) <=> vers.call(p2) }.last
-    %Q(DEVELOPER_DIR="#{latest_xcode_version}/Contents/Developer")
   end
-  private_class_method :version_select
```

Comment:
Yes, `version_select` is still a `def`'d method.

`@@version_select` is the property caching its result, but the method is still there.
It's like if in Swift you had
```swift
func version_select() -> String {
  if let cached_vs = cached_vs { return cached_vs }
  … compute result
  cached_vs = result
}
```
`version_select` is still a method, using a property named `cached_vs` / `@@version_select` to store its cached value.

---

Idea: what we could do instead to make that code less complex is split that in two:

```ruby
def self.compute_developer_dir
  # Same content as the version_select without memoizing, so basically what you have in that `begin…end` block
end

def self.version_select
  @@version_select ||= self.compute_developer_dir
end
```
Not sure what you prefer (but at least that would allow us an early in `self.compute_developer_dir`)